src/settings/index.tsx

```diff
@@ -164,6 +167,44 @@ class SettingsForm extends React.PureComponent<Props, State> {
               DOWNLOAD_ONLY_PROTOCOLS.join(", "),
             ])}
           />
+
+          <SettingsListCheckbox
+            checked={this.props.extensionState.settings.torrentTrackers.enablePublicTrackers}
+            onChange={() => {
+              this.setTorrentTrackers(
+                "enablePublicTrackers",
+                !this.props.extensionState.settings.torrentTrackers.enablePublicTrackers,
+              );
+            }}
+            label={browser.i18n.getMessage(
+              "Automatically_add_trackers_to_new_tasks_form_a_trackslist_URL",
+            )}
+          />
+
+          <li>
+            <span className="indent" />
+            <input
+              type="text"
+              {...disabledPropAndClassName(
+                !this.props.extensionState.settings.torrentTrackers.enablePublicTrackers,
+              )}
+              style={{ flex: 1 }}
+              value={this.state.publicTrackerURL}
+              onChange={(e) => {
```

Comment:
Something in this chain of behavior needs to be debounced or otherwise delayed; as currently written, I think this is going to trigger a new request to load the tracker list on every keystroke, even while the URL is only partially written.

Reply:
how about "onblur" event?

Reply:
I'm not sure how reliably onblur fires, especially if you do something like close the settings page without clicking outside of the input box. I'm imagining one of two solutions here:

1. Debounce requests to the provided URL (you can use `lodash-es/debounce`), which doesn't involve any changes to the UI.
2. Add a button to explicitly "save settings and fetch list", sort of like that which exists for the host/port/username/password configuration section. Changes to the UI required, but no clever debouncing or the like.

Having written these two down, I think the second is preferable: it's clearer about what's happening to the user, it follows existing UI patterns and it doesn't require much cleverness.